addressableled/robot.py

```diff
@@ -0,0 +1,50 @@
+from wpilib import run, TimedRobot, AddressableLED
+
+kLEDBuffer = 60
+
+class MyRobot(TimedRobot):
+    def robotInit(self):
+        # PWM Port 9
+        # Must be a PWM header, not MXP or DIO
+        self.m_led = AddressableLED(9)
+        
+        # LED Data
+        self.m_ledData = []
+
+        # Store what the last hue of the first pixel is
+        self.m_rainbowFirstPixelHue = 0
+
+        # Default to a length of 60, start empty output
+        # Length is expensive to set, so only set it once, then just update data
+        self.m_led.setLength(kLEDBuffer)
+        
+        self.m_led.start()
+
+    def robotPeriodic(self):
+        # Fill the buffer with a rainbow
+        self.rainbow()
+        
+        # Set the LEDs
+        if len(self.m_ledData) <= kLEDBuffer:
```

Comment:
Why is this check necessary? This doesn't seem to exist in the Java or C++ examples.

Reply:
I am using this check because setData argument method requests an array with a length equal to or less than the buffer.
In Java examples, it is only requested without an array, so it won't be necessary to check before.
But this was initially the form that understood the Java example.

Reply:
By the way, I'm a beginner.

Reply:
In the C++ examples, they have a [fixed size array](https://github.com/wpilibsuite/allwpilib/blob/cb33bd71dfdfa9444c372e5a0141a3e6f298a4e5/wpilibcExamples/src/main/cpp/examples/AddressableLED/cpp/Robot.cpp#L17), but that'll be challenging to do in python.. and Java too, which is probably why they created the AddressableLEDBuffer class. 

I would initialize `ledData` like so:

```python
self.ledData = [AddressableLED.LEDData() for i in range(kLEDBuffer)]
```

Then later on you can just access `self.ledData[i]` instead of appending. Also removes the need for the length check.

Reply:
Cool!